src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -860,48 +873,136 @@ public PropertySpec(string propertyName, PropertySpec? next = null)
                 /// </summary>
                 private class PropertyFetch
                 {
-                    protected PropertyFetch(Type type)
+                    public PropertyFetch(Type? type)
                     {
-                        Debug.Assert(type != null);
                         Type = type;
                     }
 
-                    internal Type Type { get; }
+                    /// <summary>
+                    /// The type of the object that the property is fetched from. For well-known static methods that
+                    /// aren't actually property getters this will return null.
+                    /// </summary>
+                    internal Type? Type { get; }
 
                     /// <summary>
-                    /// Create a property fetcher from a .NET Reflection PropertyInfo class that
-                    /// represents a property of a particular type.
+                    /// Create a property fetcher for a propertyName
                     /// </summary>
-                    public static PropertyFetch? FetcherForProperty(Type type, PropertyInfo? propertyInfo)
+                    public static PropertyFetch? FetcherForProperty(Type? type, string propertyName)
```

Comment:
Will this method ever return null?  It seems like we're either manually new'ing up an instance (in which case it obviously won't be null), or we're calling Activator.CreateInstance with a type that represents a reference type (in which case it also won't return null).  CreateInstance is annotated to return `object?` because it might be used with a nullable value type, in which case it can return null.  If this method won't actually return null, it'd be better to change the signature accordingly and then use `!` on the CreateInstance calls that won't actually return null.

Reply:
Agreed, I don't see any path this returns null.